### PR TITLE
Unitest

### PR DESCRIPTION
**Pull request by AGrigis**
- fix unitests.
- configure travis ci to run those tests with python 2.6 - 2.7.

### CODE REVIEW

**Comment by denisri** [comment]
I'm trying to look at this pull request, and it appears that it mixes several things:
- make tests pass (which is the title of the PR)
- setup travis things, which I don't know anything about, and maybe needs a little discussion
- remove everything in capsul/wip/, which also deserves a little dicsussion

(and possibly other things that I haven't seen)

Trying to run tests with the code, it appears that Capsul tests actually pass (very quickly - do they all run - I think I remember a few of them took a few seconds ?), but Morphologist tests using capsul do not, and it is a bit difficult to catch what has changed at first sight - something in study_config / FOM apparently.
So we may need a bit of help for this.

Denis

**Comment by AGrigis** [comment]
1- Travis CI is an open-source hosted, distributed continuous integration service used to build and test projects hosted at GitHub. The documentation is here http://docs.travis-ci.com/user/languages/python/.

2- In the WIP I have removed essentially replicates.

3- I can't help you with Morphologist without the full error message.

Antoine

File: capsul/study_config/config_modules/fom_config.py [hunk 1]
```diff
@@ -57,8 +58,9 @@ def initialize_module(self):
             ('output', self.study_config.output_fom),
             ('shared', self.study_config.shared_fom))
         for fom_type, fom_filename in foms:
-            fom = soma_app.fom_manager.load_foms(fom_filename)
-            self.study_config.modules_data.foms[fom_type] = fom
+            if os.path.isfile(fom_filename):
```
Comment (sapetnioc) [inline capsul/study_config/config_modules/fom_config.py]: fom_filename is not a complete path. This test is always false.

**Comment by AGrigis** [comment]
Bug with soma-workflow: need to release the project (in progress).
Do not merge this pull request.

**Comment by denisri** [comment]
This error with soma-workflow seems strange to me, and I suspect it is linked to the travis account or system confifuration.

Running on a local machine without soma-workflow configuration file, it works for me.

The error is (from travis log):

======================================================================
ERROR: test_atomic_execution (capsul.study_config.test.test_soma_workflow.TestSomaWorkflow)
----------------------------------------------------------------------
Traceback (most recent call last):
    File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/capsul_core-0.0.1-py2.7.egg/capsul/study_config/test/test_soma_workflow.py", line 194, in test_atomic_execution
        self.study_config.run(self.atomic_pipeline)
    File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/capsul_core-0.0.1-py2.7.egg/capsul/study_config/study_config.py", line 259, in run 
        workflow)
    File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/capsul_core-0.0.1-py2.7.egg/capsul/pipeline/pipeline_workflow.py", line 506, in local_workflow_run
        controller = swclient.WorkflowController(localhost)
    File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/soma_workflow/client.py", line 111, in __init__
        self.config = configuration.Configuration.load_from_file(resource_id)
    File "/home/travis/miniconda/envs/testenv/lib/python2.7/site-packages/soma_workflow/configuration.py", line 368, in load_from_file
        "in configuration file: " + config_path)
TypeError: cannot concatenate 'str' and 'instance' objects

--

Based on the sources of soma-workflow 2.6.1 (which by the way I don't know who released on febuary 2015: there is no official tag/release in git), this means that, here, config_path is not a string. It is not None either otherwise the test on line 360 (configuration.py) would have raised a different exception. So I don't know what it is...

Anyway, we should not be in this section of the code: here we are in the case of a distant computing resource (see test on line 329 and else: on line 358).
So the test on l. 329:
    if resource_id == None or resource_id == socket.gethostname() :
is False, meaning that resource_id is neither None, nor socket.gethostname().

This resource_id is passed from capsul/pipeline/pipeline_workflow.py (506):
    localhost = getfqdn().split(".")[0]
    controller = swclient.WorkflowController(localhost)

Here the only explanation is that, on the travis test system, getfqdn().split(".")[0] and socket.gethostname() do not return the same.

Why not using socket.gethostname() in capsul pipeline_workflow.py ?

Anyway I agree that soma-workflow should be investigated and return, at best, a more comprehensive error in this case (which I don't know how to test).

**Pull request merged**